bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/Profiler.java

```diff
@@ -0,0 +1,373 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.runtime.profiler;
+
+import io.ballerina.runtime.profiler.codegen.ProfilerClassLoader;
+import io.ballerina.runtime.profiler.codegen.ProfilerMethodWrapper;
+import io.ballerina.runtime.profiler.ui.HttpServer;
+import io.ballerina.runtime.profiler.ui.JsonParser;
+import io.ballerina.runtime.profiler.util.Constants;
+import io.ballerina.runtime.profiler.util.ProfilerException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static io.ballerina.runtime.profiler.util.Constants.OUT_STREAM;
+
+/**
+ * This class is used to as the driver class of the Ballerina profiler.
+ *
+ * @since 2201.8.0
+ */
+public class Profiler {
+
+    private final long profilerStartTime;
+    private String skipFunctionString = null;
+    private String balJarArgs = null;
+    private String balJarName = null;
+    private String targetDir = null;
+    private final List<String> instrumentedPaths = new ArrayList<>();
+    private final List<String> instrumentedFiles = new ArrayList<>();
+    private final List<String> utilInitPaths = new ArrayList<>();
+    private final List<String> utilPaths = new ArrayList<>();
+    private int balFunctionCount = 0;
+    private int moduleCount = 0;
+    private final ProfilerMethodWrapper profilerMethodWrapper;
+
+    public Profiler(long profilerStartTime) {
+        this.profilerStartTime = profilerStartTime;
+        this.profilerMethodWrapper = new ProfilerMethodWrapper();
+    }
+
+    private void addShutdownHookAndCleanup() {
+        // Add a shutdown hook to stop the profiler and parse the output when the program is closed.
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                long profilerTotalTime = TimeUnit.MILLISECONDS.convert(System.nanoTime(), TimeUnit.NANOSECONDS) - profilerStartTime;
+                deleteFileIfExists(Constants.TEMP_JAR_FILE_NAME);
+                OUT_STREAM.printf("%s[6/6] Generating output...%s%n", Constants.ANSI_CYAN, Constants.ANSI_RESET);
+                Thread.sleep(100);
```

Comment:
We can remove this, right?

Reply:
why?

Reply:
What is the purpose of having `Thread.sleep(100);` here?